### PR TITLE
 Fire workflows using a tree structure

### PR DESCRIPTION
**Pull request by calebcauthon**
### Whats New
 * Ability to create "tree"-style workflow rather than only "sequence"-style workflows
 * simplify event controller a bit by removing the "sequence"-style workflow creation

### Screenshots
image

### CODE REVIEW

**Comment by darrencauthon** [comment]
You can merge this at your leisure.

**Comment by calebcauthon** [comment]
I couldn't get the /create_web endpoint to work at all -- issue opening port 80. 

My guess is you're still running this on the metal, and I'm running in docker, and thats why it works for you? 

I dont want to merge and break stuff.

**Comment by darrencauthon** [comment]
@calebcauthon I'm running under docker, I've done all development under docker.

I wonder, perhaps there is some docker setup thing on your side that prevents the call out?

**Comment by calebcauthon** [comment]
I figured this out. 2 probs:
1. you have to use full URL like 'https://www.yahoo.com'
2. the `create_web` had not been updated to include the parameter url in the step config in the same manner that `create_tree` had been working. 

I fixed #2. merging!

**Comment by calebcauthon** [comment]
Also seeing some more incongruency between the 3 ways we fire events. So I'm just gonna line those up and make sure all 3 ways work.
3 ways:

/sequence/new
/events/create_web
/events_create_with_tree

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -27,9 +27,13 @@ def create_web
...
     })
 
-    steps = [{ name: 'Apple',  type: 'WebRequest' }]
-
-    workflow = Workflow.build(steps: steps)
+    workflow = Workflow.build({
+      first_step: {
+        name: 'Apple',
+        type: 'WebRequest',
+        config: { url: '{{url}}' }
```
Comment (calebcauthon) [inline app/controllers/events_controller.rb]: Had to add `config` here

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -12,9 +12,9 @@ def create
 
     originating_event = Event.create(message: params[:message])
 
-    steps = JSON.parse(request.body.read, symbolize_names: true)
+    definition = JSON.parse(request.body.read, symbolize_names: true)
 
-    workflow = Workflow.build(steps: steps)
+    workflow = Workflow.build(definition)
```
Comment (calebcauthon) [inline app/controllers/events_controller.rb]: just removed the `steps` json key

**Pull request merged**